.github/workflows/ci.yml

```diff
@@ -0,0 +1,34 @@
+name: CI
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+  push:
+    branches:
+      - master
+  workflow_dispatch:
+
+jobs:
+  cancel-previous-runs:
+    name: Cancel Previous Runs
+    runs-on: ubuntu-latest
+    steps:
+      - uses: styfle/cancel-workflow-action@0.9.1
+        with:
+          access_token: ${{ github.token }}
+
+  unit-test:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: 2.4.10
+        bundler-cache: true # runs 'bundle install' and caches installed gems automatically
+        cache-version: 1
```

Comment:
Does `cache-version` have to be specified here?

Perhaps instead of using a hardcoded value could we not compute a SHA of the `Gemfile.lock` and use that as the cache version? That way whenever gems change cache will not be found.